src/test/java/com/marklogic/spark/writer/StreamRowsTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.spark.writer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.spark.AbstractIntegrationTest;
+import com.marklogic.spark.Options;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class StreamRowsTest extends AbstractIntegrationTest {
+    StructType struct = new StructType()
+        .add("Name", DataTypes.StringType)
+        .add("House", DataTypes.StringType);
+
+    @Test
+    void streamRowsFromCsvFile(@TempDir Path tempDir) throws TimeoutException {
+
+        newDefaultStreamWriter(newSparkSession(), "src/test/resources/inputForStream", struct, "csv")
+            .outputMode("append")
+            .option(Options.WRITE_COLLECTIONS, "my-test-data")
+            // checkpointLocation needs to be a new location everytime job is run else docs are not written.
+            .option("checkpointLocation", tempDir.toFile().getAbsolutePath()).start().processAllAvailable();
+        final int expectedCollectionSize = 9;
+        String uri = getUrisInCollection("my-test-data", expectedCollectionSize).get(0);
+        JsonNode doc = readJsonDocument(uri);
+        assertTrue(doc.has("Name"));
+        assertTrue(doc.has("House"));
+    }
+
+    @Test
+    void inValidTransform(@TempDir Path tempDir) {
+        try{
+            newDefaultStreamWriter(newSparkSession(), "src/test/resources/inputForStream", struct, "csv")
+                .outputMode("append")
+                .option(Options.WRITE_COLLECTIONS, "my-test-data")
+                .option(Options.WRITE_TRANSFORM_NAME, "this-doesnt-exist")
+                .option("checkpointLocation", tempDir.toFile().getAbsolutePath()) // checkpointLocation needs to be a new location everytime job is run else docs are not written.
+                .start().processAllAvailable();
+        } catch (Exception ex){
```

Comment:
Coincidentally, we have an error on the CSV test in Jenkins - http://jenkins1.marklogic.com:9000/blue/organizations/jenkins/marklogic-spark-connector/detail/PR-25/7/tests - and the `assertTrue(doc.has("House"));` call is failing - but we don't know why. That's where something like `assertTrue(doc.has("House"), "Unexpected JSON: " + doc)` would help as we'd be able to see what the doc looks like.

Reply:
My bet is that since this is grabbing a random JSON doc from the collection, it's getting the latest one that has a name but no house. I'll make the test 100% reliable and a little more fun by using the new uriTemplate feature, and then the test can ask for a specific URI that it knows has both columns.